### PR TITLE
put back info of markdown nodes in tocnode

### PR DESCRIPTION
**Pull request by jckr**
This implements what's described in https://github.com/uber-web/ocular/issues/101. 

Currently, in luma.gl, table-of-contents.json doesn't match the contents of the folders. As a result there are some mismatches, for instance all the loaders docs have moved to loaders.gl. 

But besides that, the TOC has all the entries even "the ones that got away" 🎉

### CODE REVIEW

File: modules/gatsby/src/gatsby-node/process-nodes-markdown.js [hunk 1]
```diff
@@ -105,24 +124,39 @@ module.exports.processNewMarkdownNode = function processNewMarkdownNode({ node,
...
-
-  docNodes[relPath] = node;
-}
+  if (tocNode) {
+    // this means toc node has been created. Any markdown file processed beyond this point wouldn't have its info
+    // in the toc.
+    // but we can inject it afterwards
+    const nodeToEdit = parseToc([tocNode], relPath);
+    if (nodeToEdit) {
+      nodeToEdit.childMarkdownRemark = {
```
Comment (jckr) [inline modules/gatsby/src/gatsby-node/process-nodes-markdown.js]: the regular toc node generation process adds the full content of each markdown node to the toc. we don't need as much. The app will only use the title and slug of the corresponding markdown node for each toc entry.

File: modules/gatsby/src/gatsby-node/process-nodes-json.js [hunk 2]
```diff
@@ -9,6 +9,7 @@ module.exports.processNewDocsJsonNode = function processNewDocsJsonNode({
 }, docNodes) {
   traverseTableOfContents(node.chapters, docNodes, 1);
   tableOfContents = node;
+  return node;
```
Comment (ibgreen) [inline modules/gatsby/src/gatsby-node/process-nodes-json.js]: Looks like you left code after the return statement... can't imagine the linter being happy with this...

File: modules/gatsby/src/gatsby-node/process-nodes-markdown.js [hunk 3]
```diff
@@ -1,37 +1,37 @@
...
   let processed = false;
 
   if (!processed) {
     switch (node.internal.mediaType) {
-    case `text/markdown`:
-    case `text/x-markdown`:
-      addSourceInstanceName(...arguments);
-      processed = true;
-      break;
+      case `text/markdown`:
```
Comment (ibgreen) [inline modules/gatsby/src/gatsby-node/process-nodes-markdown.js]: I have said this before, but I think we need to get your new editor setup so it reads formatting rules from files in the repo, better avoid unrelated formatting changes. Time to implement prettier.

File: modules/gatsby/src/gatsby-node/process-nodes-markdown.js [hunk 4]
```diff
@@ -43,11 +43,30 @@ module.exports.cleanupMarkdownNode = function cleanupMarkdownNode(
   //   console.warn('Ocular ignoring node',
   //     node.absolutePath, node.internal.mediaType, node.internal.type, node.sourceInstanceName);
   // }
+};
+
+function parseToc(queue, entry) {
+  // this function returns a node in the TOC that has an entry corresponding to
```
Comment (ibgreen) [inline modules/gatsby/src/gatsby-node/process-nodes-markdown.js]: Nit: Comment that relates to whole function goes above the `function` keyword

File: modules/gatsby/src/gatsby-node/process-nodes-markdown.js (hunk 1 above)
Comment (ibgreen) [reply modules/gatsby/src/gatsby-node/process-nodes-markdown.js]: Suggestion: When I feel the need to add a comment like that to the PR to guide the reader, I usually realize that I would have been better of just adding it to the code so that future maintainers also benefit...

File: modules/gatsby/src/gatsby-node/process-nodes-json.js (hunk 2 above)
Comment (jckr) [reply modules/gatsby/src/gatsby-node/process-nodes-json.js]: ok I took a pass on that file to make it more linter friendly

File: modules/gatsby/src/gatsby-node/process-nodes-markdown.js (hunk 3 above)
Comment (jckr) [reply modules/gatsby/src/gatsby-node/process-nodes-markdown.js]: i'm all about settling for common prettier rules. we have a .eslintrc.json file in the repo but no .prettierrc or equivalent

Comment (ibgreen) [reply modules/gatsby/src/gatsby-node/process-nodes-markdown.js]: In the same repo we have a module ocular-dev-tools, I believe it includes prettier in its scripts, we should update.
